src/PowerShellEditorServices/Session/RunspaceDetails.cs

```diff
@@ -0,0 +1,182 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.CSharp.RuntimeBinder;
+using Microsoft.PowerShell.EditorServices.Utility;
+using System.Management.Automation.Runspaces;
+
+namespace Microsoft.PowerShell.EditorServices.Session
+{
+    /// <summary>
+    /// Specifies the possible types of a runspace.
+    /// </summary>
+    public enum RunspaceLocale
```

Comment:
Yeah, I was trying to find a good name for it when I thought I needed a separate designation between "location" and whether the runspace was created or attached to.  Turns out that `Debug-Runspace` merely just blocks the current runspace's pipeline and hooks into the other runspace's debugger, so no runspace change occurs.

I think I'm just going to collapse those two enums back into one which has 3 values, `Local`, `LocalProcess`, and `Remote` to reflect the 3 states that I can discern via runspace changes.  I think `Create` vs `Attach` should be a debugger distinction since it really only matters for breakpoint management (cleaning up breakpoints when detaching, etc).